fondant/manifest.py

```diff
@@ -189,11 +189,20 @@ def remove_subset(self, name: str) -> None:
         del self._specification["subsets"][name]
 
     def evolve(self, component_spec: FondantComponentSpec) -> "Manifest":
-        """Evolve the manifest based on the provided component spec. The resulting manifest is
-        the expected result if the current manifest is provided to the component defined by the
-        component spec."""
+        """Evolve the manifest based on the component spec. The resulting
+        manifest is the expected result if the current manifest is provided to the component defined
+        by the component spec."""
         evolved_manifest = self.copy()
 
+        # Update `component_id` of the metadata
+        component_id = component_spec.name.lower().replace(" ", "_")
```

Comment:
can we take this from the `image` field directly?

Reply:
the `image` field currently isn't used, and should point to a URL like `ghcr.io/ml6team/image_filtering:latest`, right?

In that case, we could split the string to get the component ID from that. For now I just used the name property.

Reply:
We use it to build the `kubeflow_component.yaml` [example](https://github.com/ml6team/fondant/blob/main/examples/pipelines/simple_pipeline/components/load_from_hub/kubeflow_component.yaml#:~:text=container%3A-,image,-%3A%20ghcr.io) the FondantSpec so it should be properly defined for each component and pointing to a valid url. 
Splitting it should be easy though so would rather do it now or add a TODO

Reply:
What benefit would using the `image` provide over using the `name`?

Reply:
you could include the docker tag of the images that you've used to run a certain a component in the manifest which can be used for versioning. (Figuring out what component version generated which manifest)